tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SQS/ReceiveMessageIntegration.cs

```diff
@@ -68,7 +76,25 @@ public class ReceiveMessageIntegration
         /// <param name="state">Calltarget state value</param>
         /// <returns>A response value, in an async scenario will be T of Task of T</returns>
         internal static CallTargetReturn<TResponse> OnMethodEnd<TTarget, TResponse>(TTarget instance, TResponse response, Exception exception, in CallTargetState state)
+            where TResponse : IReceiveMessageResponse, IDuckType
         {
+            if (response.Instance != null && response.Messages.Count > 0)
+            {
+                var dataStreamsManager = Tracer.Instance.TracerManager.DataStreamsManager;
+                if (dataStreamsManager != null && dataStreamsManager.IsEnabled)
+                {
+                    var queueName = (string)state.State;
+                    foreach (var message in response.Messages)
+                    {
+                        var adapter = new ContextPropagation.MessageAttributesAdapter(message.Attributes);
+                        state.Scope.Span.Context.MergePathwayContext(dataStreamsManager.ExtractPathwayContext(adapter));
+
+                        var edgeTags = new[] { "direction:in", $"topic:{queueName}", "type:sqs" };
+                        state.Scope.Span.SetDataStreamsCheckpoint(dataStreamsManager, CheckpointKind.Consume, edgeTags, payloadSizeBytes: 0, timeInQueueMs: 0);
```

Comment:
Struggling to understand what a data streams checkpoint is 🤔 Shouldn't we have something in `payloadSizeBytes` and `timeInQueueMs`? What are we losing because of these 0s?

Reply:
a checkpoint sends some metrics with tags, that are then used by the DSM product to show connections, number of messages and timing info.
`payloadSizeBytes` is optional, so no harm in not specifying it, but I'll see how I can compute/retrieve the payload size.
`timeInQueueMs` is not super clear to me, I'll double check the implem in the other receivers, I think I may need to write the message timestamp there